ssl/quic/quic_port.c

```diff
@@ -136,9 +140,24 @@ static int port_init(QUIC_PORT *port)
     ossl_list_port_insert_tail(&port->engine->port_list, port);
     port->on_engine_list    = 1;
     port->bio_changed       = 1;
+
+    /* Generate random key for token encryption */
+    if ((ctx = EVP_CIPHER_CTX_new()) == NULL
+        || (cipher = EVP_CIPHER_fetch(port->engine->libctx,
+                                      "AES-256-GCM", NULL)) == NULL
+        || !EVP_EncryptInit_ex(ctx, cipher, NULL, NULL, NULL)
+        || (key_len = EVP_CIPHER_CTX_get_key_length(ctx)) == 0
```

Comment:
EVP_CIPHER_CTX_get_key_length can return 0 or < 0 in an error case, you should probably check for <= 0 here

Reply:
Because key_len is currently a size_t which is unsigned, should I change it to an int?

Reply:
hmm, thats a tricky one.  It makes sense for key_len to be a size_t from a philosophical stand point, but since EVP_CIPHER_CTX_get_key_length returns an int for the size, we cant expect the key length to ever be more than INT_MAX anyway, and since we need to check for the < 0 error conditions, I would say yes, key_len should probably be an int type that you check for <=0 zero before using it for any encoding purposes